sdk/template/Azure.Template/src/Generated/MultiVersionClient.cs

```diff
@@ -0,0 +1,237 @@
+        /// <returns> The response returned from the service. </returns>
+        /// <example>
+        /// This sample shows how to call ServiceV1OperationAsync and parse the result.
+        /// <code><![CDATA[
+        /// var credential = new DefaultAzureCredential();
+        /// var endpoint = new Uri("<https://my-service.azure.com>");
+        /// var client = new MultiVersionClient(endpoint, credential);
+        /// 
+        /// Response response = await client.ServiceV1OperationAsync();
+        /// 
+        /// JsonElement result = JsonDocument.Parse(response.ContentStream).RootElement;
+        /// Console.WriteLine(result.ToString());
+        /// ]]></code>
+        /// </example>
+        public virtual async Task<Response> ServiceV1OperationAsync(RequestContext context = null)
+        {
+            using var scope = ClientDiagnostics.CreateScope("MultiVersionClient.ServiceV1Operation");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateServiceV1OperationRequest(context);
+                return await _pipeline.ProcessMessageAsync(message, context).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
+        /// <returns> The response returned from the service. </returns>
+        /// <example>
+        /// This sample shows how to call ServiceV1Operation and parse the result.
+        /// <code><![CDATA[
+        /// var credential = new DefaultAzureCredential();
+        /// var endpoint = new Uri("<https://my-service.azure.com>");
+        /// var client = new MultiVersionClient(endpoint, credential);
+        /// 
+        /// Response response = client.ServiceV1Operation();
+        /// 
+        /// JsonElement result = JsonDocument.Parse(response.ContentStream).RootElement;
+        /// Console.WriteLine(result.ToString());
+        /// ]]></code>
+        /// </example>
+        public virtual Response ServiceV1Operation(RequestContext context = null)
+        {
+            using var scope = ClientDiagnostics.CreateScope("MultiVersionClient.ServiceV1Operation");
+            scope.Start();
+            try
+            {
+                using HttpMessage message = CreateServiceV1OperationRequest(context);
+                return _pipeline.ProcessMessage(message, context);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>Servicev2Operation method. Added in service version 2022-02-02. </summary>
```

Comment:
Doc comment includes version added.

Reply:
What about *also* adding an attribute so that we could, for example, create an analyzer customers could use to help watch for these at dev time? The attribute would only be a marker, though. I'm not advocating that we use them at runtime - in fact, we shouldn't: reflection is slow.